JASP-Engine/JASP/R/ttestpairedsamples.R

```diff
@@ -199,150 +191,160 @@ TTestPairedSamples <- function(jaspResults, dataset = NULL, options, ...) {
+      confIntEffSize <- unlist(confIntEffSize) / sqrt(n)
+      
+      if (direction == "greater")
+        confIntEffSize[2] <- Inf
+      else if (direction == "less")
+        confIntEffSize[1] <- -Inf
+      
+      confIntEffSize <- sort(confIntEffSize)
+    }
+  }
+
+  ## same for all tests
+  p    <- as.numeric(res$p.value)
+  stat <- as.numeric(res$statistic)
+  sed  <- sd(c1 - c2) / sqrt(length(c1))
+  # num <- sqrt(sd(c1)^2 + sd(c2)^2 -  2 * cov(c1, c2))
+  # d   <- mean(c1) - mean(c2) / num
+  
+  m <- as.numeric(res$estimate)
+  ciLow <- ifelse(direction == "less", -Inf,
+                  as.numeric(res$conf.int[1]))
+  ciUp  <- ifelse(direction == "greater", Inf,
+                  as.numeric(res$conf.int[2]))
+  ciLowEffSize <- as.numeric(confIntEffSize[1])
+  ciUpEffSize  <- as.numeric(confIntEffSize[2])
+  
+  ## paired t-test has it, wilcox doesn't!
+  df  <- ifelse(is.null(res$parameter), "", as.numeric(res$parameter))
+  sed <- ifelse(is.null(res$parameter), "", sed)
+  
+  # add things to the intermediate results object
+  result <- list(df = df, p = p, md = m, d = d,
+                     lowerCIlocationParameter = ciLow, upperCIlocationParameter = ciUp, 
+                     lowerCIeffectSize = ciLowEffSize, upperCIeffectSize = ciUpEffSize,
+                     sed = sed)
+  
+  result[testStat] <- stat
+  
+  if (options$VovkSellkeMPR)
+    result[["VovkSellkeMPR"]] <- .VovkSellkeMPR(p)
+  
+  return(result)
+}
+
+.ttestPairedNormalFill <- function(table, dataset, options) {
   pairs <- options$pairs
-  if (!ready)
-    pairs[[1]] <- list(".", ".")
   for (pair in pairs) {
-    if(length(pair) < 2) next
+
+    if (length(pair) < 2 || pair[[1]] == pair[[2]])
+      next
+    
     p1 <- pair[[1]]
     p2 <- pair[[2]]
+  
+    row     <- list(v1 = p1, sep = "-", v2 = p2)
+    rowName <- paste(p1, p2, sep = "-")
     
-    errors <- .hasErrors(dataset,
-                         message = 'short',
-                         type = c('observations', 'variance', 'infinity'),
-                         all.target = c(p1, p2), 
-                         observations.amount  = c('< 2', '>5000'))
-    if(!identical(errors, FALSE)) {
-      container[["ttestNormalTable"]]$addFootnote(errors$message, colNames = c("W", "p"), rowNames = paste(p1, p2, sep = "-"))
-      container[["ttestNormalTable"]]$addRows(list(v1 = p1, sep = "-", v2 = p2, W = "NaN", p = "NaN"), rowNames = paste(p1, p2, sep = "-"))
-      next
-    }
-    if (ready && p1 != p2) {
-      c1   <- dataset[[ .v(p1) ]]
-      c2   <- dataset[[ .v(p2) ]]
-      data <- na.omit(c1 - c2)
+    if (p1 != "" && p2 != "") {
+      errors <- .hasErrors(dataset,
+                           message = 'short',
+                           type = c('observations', 'variance', 'infinity'),
+                           all.target = c(p1, p2), 
+                           observations.amount  = c('< 2', '> 5000'))
```

Comment:
Just out of curiosity, why do we check `> 5000`?

Reply:
It's in the shapiro wilks test:
```
> stats::shapiro.test
function (x) 
{
    DNAME <- deparse(substitute(x))
    stopifnot(is.numeric(x))
    x <- sort(x[complete.cases(x)])
    n <- length(x)
    if (is.na(n) || n < 3L || n > 5000L) 
        stop("sample size must be between 3 and 5000")
    rng <- x[n] - x[1L]
    if (rng == 0) 
        stop("all 'x' values are identical")
    if (rng < 1e-10) 
        x <- x/rng
    res <- .Call(C_SWilk, x)
    RVAL <- list(statistic = c(W = res[1]), p.value = res[2], 
        method = "Shapiro-Wilk normality test", data.name = DNAME)
    class(RVAL) <- "htest"
    return(RVAL)
}
```